src/palrt/guid.cpp

```diff
@@ -24,3 +24,26 @@ DEFINE_GUID(IID_IClassFactory, 0x00000001, 0x0000, 0x0000, 0xC0, 0x00, 0x00, 0x0
 // objidl.idl
 DEFINE_GUID(IID_ISequentialStream, 0x0c733a30, 0x2a1c, 0x11ce, 0xad, 0xe5, 0x00, 0xaa, 0x00, 0x44, 0x77, 0x3d);
 DEFINE_GUID(IID_IStream, 0x0000000c, 0x0000, 0x0000, 0xC0, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x46);
+
+// Create a random guid based on the https://www.ietf.org/rfc/rfc4122.txt
+HRESULT
+PALAPI
+CoCreateGuid(OUT GUID * pguid)
+{
+    PAL_Random(TRUE /* bStrong */, pguid, sizeof(GUID));
+
+    static const USHORT VersionMask = 0xF000;
+    static const USHORT RandomGuidVersion = 0x4000;
+
+    static const BYTE ClockSeqHiAndReservedMask = 0xC0;
+    static const BYTE ClockSeqHiAndReservedValue = 0x80;
+
+    // Modify bits indicating the type of the GUID
+
+    // time_hi_and_version
+    pguid->Data3 = (pguid->Data3 & ~VersionMask) | RandomGuidVersion;
+    // clock_seq_hi_and_reserved
+    pguid->Data4[0] = (pguid->Data4[0] & ~ClockSeqHiAndReservedMask) | ClockSeqHiAndReservedValue;
```

Comment:
At one point we had an issue related to guid perf:
https://github.com/dotnet/corefx/issues/3573
Have you looked to see how this compares to the previous implementation?

Reply:
Good point, I've completely forgotten that we were using PAL_Random and had that perf issue in the past. I'll make the change recommended by @jkotas and try to make sure the perf doesn't degrade.